odoo/models.py

```diff
@@ -3320,7 +3320,8 @@ def get_base_url(self):
         :rtype: string
 
         """
-        self.ensure_one()
+        if len(self) > 1:
+            raise ValueError("Expected singleton or no record: %s" % self)
```

Comment:
What is the point of checking for singleton or no record? It is important in the override in the website module, but here, I think it is an extra useless check